### PR TITLE
Build and deployment logs should check kubelet response

### LINKED ISSUE #5468: No error when fail to get build logs in UI
For old builds, there is no error when we fail to get the build log. The web console displays an empty log.

/cc @jwforres @benjaminapetersen

### PR DESCRIPTION
**Pull request by smarterclayton**
If the kubelet responds with an error, transform that for the client

@spadgett @liggitt

Fixes #5468

### CODE REVIEW

File: pkg/build/registry/buildlog/rest.go [hunk 1]
```diff
@@ -110,10 +110,11 @@ func (r *REST) Get(ctx kapi.Context, name string, opts runtime.Object) (runtime.
...
 	return &genericrest.LocationStreamer{
-		Location:    location,
-		Transport:   transport,
-		ContentType: "text/plain",
-		Flush:       buildLogOpts.Follow,
+		Location:        location,
+		Transport:       transport,
+		ContentType:     "text/plain",
+		Flush:           buildLogOpts.Follow,
+		ResponseChecker: genericrest.NewGenericHttpResponseChecker("Pod", name),
```
Comment (liggitt) [inline pkg/build/registry/buildlog/rest.go]: s/name/buildPodName/

File: pkg/deploy/registry/deploylog/rest.go [hunk 2]
```diff
@@ -144,10 +144,11 @@ func (r *REST) Get(ctx kapi.Context, name string, opts runtime.Object) (runtime.
...
 	return &genericrest.LocationStreamer{
-		Location:    location,
-		Transport:   transport,
-		ContentType: "text/plain",
-		Flush:       deployLogOpts.Follow,
+		Location:        location,
+		Transport:       transport,
+		ContentType:     "text/plain",
+		Flush:           deployLogOpts.Follow,
+		ResponseChecker: genericrest.NewGenericHttpResponseChecker("Pod", name),
```
Comment (liggitt) [inline pkg/deploy/registry/deploylog/rest.go]: s/name/deployPodName/

**Comment by smarterclayton** [comment]
[test]

**Comment by liggitt** [comment]
LGTM

**Comment by smarterclayton** [comment]
Waiting for @spadgett

**Comment by spadgett** [comment]
@smarterclayton I tested, and it fixes #5468

**Comment by smarterclayton** [comment]
[merge]

**Pull request merged**